### PR TITLE
update edge-containers to latest oras and embedded containerd 1.6.1

### PR DESCRIPTION
**Pull request by deitch**
This is half of the containerd 1.6.1 upgrade, which is good practice, but more importantly mandated by the [CVE](https://nvd.nist.gov/vuln/detail/CVE-2022-23648).

This part updates the library [edge-containers](https://github.com/lf-edge/edge-containers) to containerd 1.6.1, as well as some other dependencies. 

The second phase will be updating containerd itself in the root.

### CODE REVIEW

**Comment by deitch** [comment]
Latest versions of containerd do not work with go 1.15.x, which we are using in our builders. I bumped the go version in this PR (min 1.16 in `pkg/pillar/go.mod`, actual 1.17.7 in Makefile and go builder image). If it doesn't break anything, then we stick with it.

**Comment by deitch** [comment]
To get this to work, in addition to the containerd dependency changes and the edge-containers dependency changes, I had to:

- bump our version of go in builder Dockerfile and Makefile from 1.15 to 1.17 (I could have used 1.16, but might as well go for most modern)
- update the builder Dockerfile, which does a lot of deprecated installations

**Review by eriknordmark — APPROVED** [review]
Kick off eden again

Note tar ztests sees failures like
FailureInfo: Error OperState. Error Details: [{'description': 'error pulling 72b07c01-d83f-4e7d-ad71-60a1a8ab71ff-app-images/ubuntu-bionic-18.04.qcow2 from containerd: unsupported\n\n', 'severity': 'SEVERITY_ERROR', 'timestamp': '2022-03-14T18:54:11.571463654Z', 'entities': [{'entity': 'ENTITY_VOLUME', 'entityId': '96ecf5fe-c009-4659-9a7e-c08fe84bc165'}], 'retryCondition': ''}]

and

***Upgrade to Expected Image Failed. ErrMsg: installDownloadedObject(0745d9e0-12db-4865-b837-e2662a4e2d0b): InstallWorkResult error, exception while installing: error pulling 0745d9e0-12db-4865-b837-e2662a4e2d0b-images/AAFlABDe6Nm63ixF5LJBbk9Keqp6/5.21.8-kvm-amd64 from containerd: unsupported

So something is amiss. Perhaps an eden run will also show failures.

**Comment by deitch** [comment]
Thanks, let's keep an eye on this. There were some upstream changes, so it is possible we triggered something (that _is_ what CI is there for). If it happens again, can you link to the logs here?

File: pkg/pillar/cmd/domainmgr/domainmgr_test.go [hunk 1]
```diff
@@ -371,10 +371,12 @@ func TestHandleMimeMultipart(t *testing.T) {
 			expectMultipart: true,
 			expectFail:      false,
 		},
+		// go mime package is fixed to comply with RFC 7578, Section 4.2 which requires that if
+		// a filename is provided, the directory path information must not be used.
```
[the remark this answers is not in the document]
Comment (eriknordmark) [reply pkg/pillar/cmd/domainmgr/domainmgr_test.go:375]: I've added tests for the directory structure in #2554. Can you pick up that patch and see if it passes?

**Comment by eriknordmark** [comment]
Eden is also failing with
            vol_test.go:170: 		state: CREATING_VOLUME: description:"error pulling 648fa86a-0858-4b57-8ab5-93709c15ae53-eserver/empty.vmdk from containerd: unsupported" timestamp:{seconds:1647287409 nanos:683530891} severity:SEVERITY_ERROR received in: 2022-03-14T19:50:35.62406178Z

See https://github.com/lf-edge/eve/runs/5543515630?check_suite_focus=true

**Comment by deitch** [comment]
I saw #2554 ; once that is in, I can rebase this on that. In the meantime, I will check that error.

**Comment by deitch** [comment]
OK, I think I got it now. @eriknordmark let's let these run, and when green, we can kick off eden tests again.

**Review by eriknordmark — CHANGES_REQUESTED** [review]
So we need to find a way to preserve the capability to use directory path names in the mime multipart. Don't know if the easiest is to fork that code or if we can do better.

**Review by eriknordmark — CHANGES_REQUESTED** [review]
With the workaround in the mime code you can now proceed with this PR.

**Review by eriknordmark — APPROVED** [review]

**Comment by deitch** [comment]
@eriknordmark rebased on `master` now that the mime PR #2554 is in, and removed the mime test change that this PR had. It should be clean now.

Let's let regular CI run, after that is clean full eden, and if all is good, we can merge it in.

**Review by eriknordmark — APPROVED** [review]
Run eden again

**Comment by deitch** [comment]
I believe we have the issue @petr-zededa raised fixed. This needs the normal CI to be green, and then eden to run again.

**Review by giggsoff — APPROVED** [review]
Let test it

**Review by rvs — COMMENTED** [review]
LGTM

**Pull request merged**